### ISSUE TITLE
Az CLI login non-interactive with username and password throws ascii encode error 

### ISSUE THREAD

**Issue by naveenkumartd17** [Azure/azure-cli#17663]
## Describe the bug

Login to Azure using CLI  az login non-interactive throws 'ascii' codec can't encode characters in position 1617-1618: ordinal not in range(128)

**Command Name**
az login -u naveenkumar.td@** -p ****

**Errors:**

The command failed with an unexpected error. Here is the traceback:
'ascii' codec can't encode characters in position 1617-1618: ordinal not in range(128)
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 660, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 723, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 716, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\six.py", line 703, in reraise
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 694, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 331, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/__init__.py", line 811, in default_command_handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/profile/custom.py", line 162, in login
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/_profile.py", line 215, in find_subscriptions_on_login
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/_profile.py", line 885, in find_from_user_account
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/authentication_context.py", line 164, in acquire_token_with_username_password
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/authentication_context.py", line 128, in _acquire_token
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/authentication_context.py", line 162, in token_func
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/token_request.py", line 285, in get_token_with_username_password
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/token_request.py", line 258, in _get_token_username_password_federated
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/token_request.py", line 210, in _perform_username_password_for_access_token_exchange
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/token_request.py", line 197, in _perform_wstrust_exchange
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/wstrust_request.py", line 169, in acquire_token
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/wstrust_request.py", line 126, in _handle_rstr
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/wstrust_response.py", line 238, in parse
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\adal/wstrust_response.py", line 217, in _parse_token_by_re
UnicodeEncodeError: 'ascii' codec can't encode characters in position 1617-1618: ordinal not in range(128)

## To Reproduce:

1.Login using az login non interactive command. Pass username and password as arguments.

## Expected Behavior

Login should successful.

## Environment Summary
```
Windows-10-10.0.14393-SP0
Python 3.9.1
Installer: MSI

azure-cli 2.21.0 *

```
## Additional Context

az login with service principal and interactive is successful.

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by naveenkumartd17** [comment]
This issue happens in Windows server 2016

**Comment by jiasli** [comment]
Looks like a same issue as https://github.com/Azure/azure-cli/issues/17444. We fixed it in https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/251 and released it in 2.22.0. Please update to the latest version and see if the issue persists.

**Comment by naveenkumartd17** [comment]
Thank you @jiasli. I have upgraded to latest version and it is working as expected.
